flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/SourceBuilder.java

```diff
@@ -526,9 +526,12 @@ public TokenBuilder expression(String exprAsStr) {
         }
 
         public TokenBuilder expressionWithType(Property type, Property variable) {
-            String typeSourceCode = type.toSourceCode();
-            int lastDotIndex = typeSourceCode.lastIndexOf('.');
-            String typeName = lastDotIndex >= 0 ? typeSourceCode.substring(lastDotIndex + 1) : type.toSourceCode();
+            // Handle module prefixed types.
+            // Transform <prefix>.<module>:<Type> to <module>:<Type>.
+            // Example: lang.jsondata:Error -> jsondata:Error
+            String regex = "([A-Za-z]\\w*)\\.([A-Za-z]\\w*):([A-Za-z]\\w*)";
+            String typeName = type.toSourceCode().replaceAll(regex, "$2:$3");
```

Comment:
With the new client, we are implementing type validations in the form, which blocks users from saving if any validation errors exist. Therefore, we can assume that the provided type will always be correct, and pass the value as it is.

Reply:
This change is to address cases such as `lang.jsondata:Type` (with module prefixes). Are we handling them in the type validation?

Reply:
Yes

Reply:
If they are getting converted correctly to `jsondata:Type`, we don't need this change.

Reply:
If so, shall we keep only the test cases?

Reply:
Yeah.

Reply:
If user enters `data.jsondata:Error`, the form cannot be saved as there is a diagnostic error

Reply:
Removed the regex validation with [d3c0732](https://github.com/ballerina-platform/ballerina-dev-tools/pull/618/commits/d3c0732ca9b63fc8d5b1839c465492be8b8eb1cf)